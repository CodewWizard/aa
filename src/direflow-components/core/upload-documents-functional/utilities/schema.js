export const EmailDocumentSchemaUI = [
    {
      key: "EmailDocument",
      type: "Card",
      title: "EMAIL DOCUMENT",
      className: "",
      titleClassName: "commonHead commonHeadPadding",
      childsClassName: "",
      controls: [
        {
          key: "emailid",
          label: "Email Id*",
          type: "email",
          props: { required: true, class: "inputText", placeholder: "Email Id"},
          inputClass: "mb-3 flex-nowrap",
          positionClass: "col-md-12",
          isColummnDisplay: false,
          dataPath: "EmailId"
        },
        { 
          key: "optionalcomment", 
          type: "textarea inputTextArea", 
          props: { class: "inputText", rows:"4", cols:"50"}, 
          label: "Add optional email content",
          labelClass: "", 
          inputClass: "",
          positionClass: "col-md-12 mt-4", 
          isColummnDisplay: false, 
          dataPath: "Comments"
        }
      ]
    }
];

export const UploadDocumentSchema = [
    {
      key: "ManualFIlesUpload",
      type: "Card",
      title: "MANUAL UPLOAD",
      className: "",
      titleClassName: "commonHead commonHeadPadding",
      childsClassName: "",
      controls: [
        {
          key: "policynumber",
          label: "QUOTE / POLICY NUMBER*",
          props: { required: true, className: "inputText", disabled: true},
          inputClass: "mb-3 flex-nowrap",
          positionClass: "col-md-6",
          isColummnDisplay: false,
          dataPath: "PolicyNumber"
        },
        {
          key: "documentType",
          label: "DOCUMENT TYPE*",
          props: { required: true, className: "inputText", placeholder: "Document Type" },
          inputClass: "mb-3 flex-nowrap",
          positionClass: "col-md-6",
          isColummnDisplay: false,
          dataPath: "DocumentType"
        },
        { 
          key: "fileupload", 
          type: "file", 
          props: {required: true, className: "form-control", multiple: "multiple", accept: "application/pdf, application/vnd.ms-excel"}, 
          label: "SELECT FILES TO UPLOAD*",
          labelClass: "col-md-4", 
          inputClass: "",
          positionClass: "col-md-12 mt-4", 
          isColummnDisplay: false, 
          dataPath: "UploadedFiles"
        },
        {
          key: "noteslabel1",
          label: "",
          type: "label",
          value: "* Maximum file size - 25 MB",
          props: { className: "" },
          labelClass: "",
          inputClass: "col-md-8",
          positionClass: "col-sm-12 pt-3 pb-0 mb-0 fileuploadNotes",
          isColummnDisplay: false
        },
        {
          key: "noteslabel2",
          label: "",
          type: "label",
          value: "** Allowed Formats - jpeg, jpg, png, gif, bmp, xls, xlsx, doc, docx, odt, txt, pdf, pptx, ppt, rtf, eml, csv",
          props: { className: "" },
          labelClass: "",
          inputClass: "col-md-8",
          positionClass: "col-sm-12 pt-0 pb-0 mb-0 fileuploadNotes",
          isColummnDisplay: false
        }
      ]
    }
]

export const ManualFilesUploadSchemaDATA = {
  "PolicyNumber": "",
  "DocumentType": "",
  "UploadedFiles": []
};

export const EmailDocumentDataModel = {
  EmailId: "",
  Comments: "",
}