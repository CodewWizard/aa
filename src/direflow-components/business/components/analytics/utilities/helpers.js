import axios from "axios";
import { axiosKeys, consoleLogs } from "../constants/constants";
import { advanceFilterModel } from "../constants/schemas";

const callAPI = async (type, url, payload, _headers = {}, _responsetype = '') => {
  try {
    const config = {
      headers: _headers,
      responseType: _responsetype
    }
    switch (type) {
      case axiosKeys.PostAction: {
        try {
          const response = await axios.post(url, payload, config);
          return response.status === 204 ? true : response.data;
        } catch (error) {
          console.error(consoleLogs.Utilities.Shared.Helpers.AxiosPost, error);
          throw error;
        }
      }
      case axiosKeys.GetAction: {
        try {
          const response = await axios.get(url, config);
          return response.data;
        } catch (error) {
          console.error(consoleLogs.Utilities.Shared.Helpers.AxiosGet, error);
          throw error;
        }
      }
    }
  } catch (error) {
    console.error(consoleLogs.Utilities.Shared.Helpers.CallAPI, error);
    throw error;
  }
}

const removeDuplicateColumns = (table1, table2, propertyName) => {
  const filteredArray = table2.filter(obj1 =>
    !table1.some(obj2 => obj1[propertyName] === obj2[propertyName])
  );
  return filteredArray;
}

const updateFilters = (filters, filterIndex) => {
  const tempFilters = [...filters];
  let TransactionYearFilterIndex = tempFilters.findIndex(obj => obj.key === "TransactionYear")
  let TransactionMonthFilterIndex = tempFilters.findIndex(obj => obj.key === "TransactionMonth")
  if (filterIndex == TransactionYearFilterIndex && TransactionMonthFilterIndex > -1) {
    tempFilters.splice(TransactionMonthFilterIndex, 1)
  }
  tempFilters.splice(filterIndex, 1);
  return tempFilters;
}

const updateFiltersFormData = (filtersFormData, filters) => {
  let tempdata = { ...filtersFormData };
  Object.entries(filtersFormData).forEach(([k, v]) => {
    tempdata[k] = "";
  });
  filters.forEach(item => {
    tempdata[item.key] = item.value;
  })
  return tempdata;
}

const updateFiltersByKey = (filters, model, key) => {

  let tempFilters = JSON.parse(JSON.stringify(filters))
  const indexOfObject1 = tempFilters.findIndex(object => object.key == key);
  if (model?.[key]?.length > 0) {
    let item1 = { key, value: model[key] };
    if (indexOfObject1 > -1) {
      tempFilters[indexOfObject1].value = model[key];
    } else {
      tempFilters.push(item1);
    }
  } else if (indexOfObject1 > -1) {
    tempFilters.splice(indexOfObject1, 1);
  }

  return tempFilters;
}

const filterPoliciesByModel = (model, policies) => {
  let tempPolicies = [...policies];
  Object.entries(model).forEach((k) => {
    if (model[k[0]]?.length > 0) {
      tempPolicies = tempPolicies.filter(x => x[k[0]] == model[k[0]]);
    }
  })
  return tempPolicies;
}

const getFilterKey = (key) => {
  switch (key) {
    case 'TransactionYear':
      return 'Year'
    case 'TransactionMonth':
      return 'Month'
    case 'TransactionType':
      return 'Transaction'
    case 'AgencyName':
      return 'Agency'
    case 'UnderwriterName':
      return 'Underwriter'
    case 'Lob':
      return 'LOB'
    case 'RenewalFlag':
      return 'RB-NB'
    default:
      return
  }
}
const AddUWExtraFields = (role, schema) => {
  if (role.toLowerCase() == 'underwriter') {
    schema[0].controls.push(
      {
        key: "AgencyName",
        type: "search-dropdown",
        value: "",
        label: "Select Agency",
        props: { required: true, class: "inputText" },
        labelClass: "inputLabel ddlUnderwriterList",
        positionClass: "col-xxl-3 col-lg-4 col-md-2",
        isColummnDisplay: false,
        dataPath: "AgencyName",
        options: []
      },
      {
        key: "UnderwriterName",
        value: "",
        label: "Select UW",
        type: "search-dropdown",
        props: { class: "inputText", placeholder: "testing uw" },
        labelClass: "inputLabel ddlUnderwriterList",
        positionClass: "col-xxl-3 col-lg-4 col-md-2",
        isColummnDisplay: false,
        dataPath: "UnderwriterName",
        options: []
      },
    )
  }
  return schema;
}

const GetSchema = (schema = false, role) => {
  let filterSchema = advanceFilterModel;
  if (schema) {
    return schema;
  }
  filterSchema = AddUWExtraFields(role, advanceFilterModel);
  return filterSchema;
}

const SetSchemaDefaultValues = (defaultSchema, props) => {
  let schema = defaultSchema;
  schema = JSON.parse(JSON.stringify(schema));
  schema.map(s => {
    s.controls = s.controls.map(c => {
      switch (c.key) {
        case 'AgencyName':
          c.options.push(...props.agencyList)
          c.props.primaryColor = props.primaryColor
          break;
        case 'UnderwriterName':
          c.options.push(...props.uwList)
          c.props.primaryColor = props.primaryColor
          break;
        case 'State':
          c.options.push(...props.stateList)
          c.props.primaryColor = props.primaryColor
          break;
        case 'Lob':
          c.options.push(...props.lobsList)
          c.props.primaryColor = props.primaryColor
          break;
      }
      return c;
    });
    return s;
  })

  if (props.setDefaultValuesHandler) {
    const tempSchema = props.setDefaultValuesHandler(schema);
    if (tempSchema)
      schema = tempSchema;
  }
  return schema;
}

const GetColor = (colorArray) => {
  if (colorArray.length == 1) return colorArray
  let k = colorArray.length;
  colorArray.splice(0, k - 1)
  return colorArray
}

const GroupByKey = (array, key) => {
  return array
    .reduce((hash, obj) => {
      if (obj[key] === undefined) return hash;
      return Object.assign(hash, { [obj[key]]: (hash[obj[key]] || []).concat(obj) })
    }, {});
}

// const GroupByKeyUW = (array, key) => {
//   return array.reduce((hash, obj) => {
//     if (obj[key] === undefined) return hash;
//     const keyValue = obj[key];
//     if (!hash[keyValue]) {
//       hash[keyValue] = { count: obj.count, data: [obj] };
//     } else {
//       hash[keyValue].count += obj.count;
//       hash[keyValue].data.push(obj);
//     }
//     return hash;
//   }, {});
// };

// const GroupByKeyUW = (array, key) => {

//   array.sort((a, b) => new Date(`${a.TransactionMonth}-01-${a.TransactionYear}`) < new Date(`${b.TransactionMonth}-01-${b.TransactionYear}`) ? 1 : -1);
//   const uniqueObjects = array.filter(
//     (array, index, self) =>
//       index ===
//       self.findIndex((t) => t.QuoteNumber === array.QuoteNumber && t.TransactionType === array.TransactionType)
//   );
//   let result = GroupByKeyWithSum(uniqueObjects, key);

//   let keywisecount = [];
//   Object.entries(result).forEach(item => {
//     keywisecount.push({ label: item[0], count: item[1].count });
//   });
//   return keywisecount;
// };

// const GroupByKeyWithSum = (array, key) => {
//   const result = {};
//   array.forEach(obj => {
//     const keyValue = obj[key];
//     if (!result[keyValue]) {
//       if (obj.IsPreBind) {

//         result[keyValue] = { count: 1 }
//       } else {
//         result[keyValue] = { count: obj.count };
//       }
//     } else {
//       if (obj.IsPreBind) {
//         result[keyValue].count += 1
//       } else {
//         result[keyValue].count += obj.count;
//       }
//     }
//   });
//   return result;
// };


const GroupByKeyUW = (array, key) => {
  // No sorting if not necessary
  // array.sort((a, b) => new Date(`${b.TransactionMonth}-01-${b.TransactionYear}`) - new Date(`${a.TransactionMonth}-01-${a.TransactionYear}`));
  // const uniqueObjects = filterUniqueObjects(array);
  const result = groupByKeyWithSum(array, key);
  return transformResult(result);
};

const filterUniqueObjects = (array) => {
  const seen = new Set();
  const uniqueObjects = [];
  for (let obj of array) {
    const key = `${obj.QuoteNumber}-${obj.TransactionType}`;
    if (!seen.has(key)) {
      seen.add(key);
      uniqueObjects.push(obj);
    }
  }
  return uniqueObjects;
};

const filterUniqueVersions = (array) => {
  array.sort((a, b) => parseInt((b.QuoteVersion)) - parseInt(a.QuoteVersion));
  const seen = new Set();
  const uniqueObjects = [];
  for (let obj of array) {
    const key = obj.QuoteNumber;
    if (!seen.has(key)) {
      seen.add(key);
      uniqueObjects.push(obj);
    }
  }
  return uniqueObjects;
};

const groupByKeyWithSum = (array, key) => {
  const result = {};
  for (let obj of array) {
    const keyValue = obj[key];
    if (!result[keyValue]) {
      result[keyValue] = { count: 0 };
    }
    result[keyValue].count += obj.IsPreBind ? 1 : obj.count;
  }
  return result;
};

const transformResult = (result) => {
  const keywisecount = [];
  for (let [key, value] of Object.entries(result)) {
    keywisecount.push({ label: key, count: value.count });
  }
  return keywisecount;
};


const CountGroupItems = (obj) => {

  let keywisecount = [];
  Object.entries(obj).forEach(item => {
    keywisecount.push({ label: item[0], count: item[1].length });
  });
  return keywisecount;
}

const GetUniqueListBy = (arr, key) => {
  return [...new Map(arr.map(item => [item[key], item])).values()]
}

export { CountGroupItems, GetColor, GetSchema, GetUniqueListBy, GroupByKey, GroupByKeyUW, SetSchemaDefaultValues, callAPI, filterPoliciesByModel, filterUniqueVersions, getFilterKey, removeDuplicateColumns, updateFilters, updateFiltersByKey, updateFiltersFormData };


