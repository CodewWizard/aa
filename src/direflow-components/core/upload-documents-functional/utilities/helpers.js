import axios from "axios";
import moment from "moment";
import { v4 } from "uuid";

const callAPI = async (type, url, payload, _headers = {}, _responsetype = "") => {
  try {
    const config = {
      headers: _headers,
      responseType: _responsetype,
    };
    switch (type) {
      case "POST": {
        try {
          const response = await axios.post(url, payload, config);
          return response.status === 204 ? true : response.data;
        } catch (error) {
          console.error(
            "consoleLogs.Utilities.Shared.Helpers.AxiosPost",
            error
          );
          throw error;
        }
      }
      case "GET": {
        try {
          const response = await axios.get(url, config);
          return response.data;
        } catch (error) {
          console.error("consoleLogs.Utilities.Shared.Helpers.AxiosGet", error);
          throw error;
        }
      }
    }
  } catch (error) {
    console.error("consoleLogs.Utilities.Shared.Helpers.CallAPI", error);
    throw error;
  }
};

export const getFilesListFromDMS = async (requestInput, apiBaseUrl, dmsAuthApi) => {
  try {
    let apiDynamicURL = "";
    Object.entries(requestInput).forEach((entry, index) => {
      const [key, value] = entry;
      apiDynamicURL = apiDynamicURL + key + "=" + value;
      if (key != "cloudBlobContainer") {
        apiDynamicURL = apiDynamicURL + "&";
      }
    });
    const apiURL = apiBaseUrl + apiDynamicURL;
    const apiResponse = await callAPI("GET", apiURL, null, {Authorization: dmsAuthApi}, "");
    return apiResponse;
  } catch (err) {
    console.error("consoleLogs.Utilities.Pages.Application.GetFilesListFromDMS", err);
    throw err;
  }
};

const TimeFormattingAmPm = (value) => {
  let time;
  if (value == "00:00") time = "12:01 AM";
  else {
    var newTime = value.split(":");
    var hours = newTime[0];
    var minutes = newTime[1];
    var timehours = parseInt(hours) > 12 ? (parseInt(hours) % 12).toString().padStart(2, "0") : parseInt(hours).toString().padStart(2, "0");
    var timeminutes = parseInt(minutes) > 9 ? parseInt(minutes).toString().padStart(2, "0") : parseInt(minutes).toString().padStart(2, "0");
    time = timehours + ":" + timeminutes + (parseInt(hours) < 12 ? " AM" : " PM");
  }
  return time;
};

export const uploadFilesToDMS = async (policy, filesToUpload, userID, apiBaseURL, dmsAuthApi, dmsDefaultCoverHolder, dmsApiName) => {
  try {
    let date = moment().utc(false).format("YYYY-MM-DD");
    date = moment().format("YYYY-MM-DD");

    let dtString = date.split("-");

    let timeString = TimeFormattingAmPm(moment().utc(false).format("HH:mm:ss"));
    let tempFilesArray = [...filesToUpload.UploadedFiles];

    if (tempFilesArray.length > 0) {
      const infoId = policy.PolicyNumber
        ? policy.PolicyNumber
        : policy.QuoteNumber;
      let promises = tempFilesArray.map(async (file) => {
        const metadata = {
          PolicyInfoId: infoId,
          PolicyNumber: policy.PolicyNumber,
          DocumentType: `${filesToUpload.DocumentType}_${v4()}`,
          CarrierCode: policy.Attributes.Carrier,
          CoverholderCode: policy.Attributes.Coverholder,
          Lob: policy.Attributes.Lob,
          State: policy.Attributes.State,
          TransactionCount:
            policy.Transaction.Number && policy.Transaction.Number >= 0
              ? policy.Transaction.Number?.toString()
              : "0",
          TransactionType: policy.Transaction.Type,
          TransactionDate: `${dtString[1].padStart(2, 0)}-${dtString[2].padStart(2, 0)}-${dtString[0]}, ${timeString}`,
          UserName: userID,
          Filename: `${infoId}_${policy.Transaction.Type}_${date}_${timeString}.pdf`,
          Applicationstatus: policy.PolicyStatus,
          CreatedDateTime: `${dtString[1].padStart(2, 0)}-${dtString[2].padStart(2, 0)}-${dtString[0]}, ${timeString}`,
        };

        let formData = new FormData();
        formData.append("file", file);
        formData.append("cloudblobcontainer", dmsApiName);
        formData.append("carriercode", policy.Attributes.Carrier);
        formData.append("coverholdercode", policy.Attributes.Coverholder || dmsDefaultCoverHolder);
        formData.append("lobcode", policy.Attributes.Lob);
        formData.append("statecode", policy.Attributes.State);
        formData.append("referenceNumber", infoId);
        formData.append("metadata", JSON.stringify(metadata));
        formData.append("filename", metadata.Filename);

        const apiURL = apiBaseURL;
        const apiResponse = await callAPI("POST", apiURL, formData, {"Authorization": dmsAuthApi, "content-type": "multipart/form-data"});
        return apiResponse;
      });

      return await Promise.all(promises).then((resp) => {
        return resp;
      });
    }
    return false;
  } catch (error) {
    console.error("consoleLogs.Utilities.Pages.Application.UploadFilesToDMS", error);
    throw error;
  }
};