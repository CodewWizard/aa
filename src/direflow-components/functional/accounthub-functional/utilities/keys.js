const dateKeys = {
    Seconds: "Seconds",
    Minutes: "Minutes",
    Hours: "Hours",
    Days: "Days",
    Weeks: "Weeks",
    Months: "Months",
    Years: "Years",
    DateFormatyyymmdd: "yyyy-mm-dd"
}

const commonKeys = {
    Code: "Code",
    Description: "Description",
    Value: "Value",
    Text: "Text",
    IsErrored: "IsErrored",
    Valid: "Valid",
    Errors: "Errors",
    EmptyString: "",
    DefaultTemplate: "Default Template",
    ModuleQuote: "Quote",
    ModuleApplication: "Application",
    ModuleEndorsement: "Endorsement",
    ModuleRenewal: "Renewal",
    ModuleCancel: "Cancel",
    ModuleReinstate: "Reinstate",
    RoleAgent: "Agent",
    Agency: "Agency",
    RoleUW: "UW",
    RoleUnderwriter: "Underwriter",
    RoleSrUnderwriter: "Sr. Underwriter",
    RoleUnderwritingManager: "Underwriting Manager",
    RoleAdmin: "Admin",
    RoleMGA: "MGA",
    RoleCarrier: "Carrier",
    RoleAuditor: "Auditor",
    RoleSUW: "SUW",
    ActionAllow: "Allow",
    ActionRefer: "Refer",
    ActionBlock: "Block",
    ActionInEligible: "InEligible",
    ActionEmailQuote: "EmailQuote",
    ActionEmailFormalQuote: "EmailFormalQuote",
    ActionPrintQuote: "PrintQuote",
    ActionDownloadRatesheet: "DownloadRatesheet",
    ActionSmsQuote: "SmsQuote",
    ActionEmailAndDownload: "ActionEmailAndDownload",
    NotificationTypeSMS: "S",
    NotificationTypeEmail: "E",
    NotificationFormatHTML: "html",
    NotificationFormatPDF: "pdf",
    NotificationFormatTEXT: "text",
    TemplateNameNotificationQuote: "toga/commercial/auto/common/ca_email.html",
    TemplateNameNotificationFormalQuote: "toga/commercial/auto/common/formalquote.html",
    TemplateNameNotificationFormalQuoteMessageBody: "toga/commercial/auto/common/formalquotemessagebody.html",
    TemplateNameCancelReinstate: "toga/commercial/auto/common/REI_CAN_document.html",
    TemplateGLRateSheet: "toga/templates/commercial/auto/common/ratesheets/Canopius General Liabilty Rater.html",
    TemplateALRateSheet: "toga/templates/commercial/auto/common/ratesheets/Alpine Liability Rater.html",
    TemplateMTGRateSheet: "toga/templates/commercial/auto/common/ratesheets/Canopius Motor Truck Cargo Rater.html",
    TemplatePDRateSheet: "toga/templates/commercial/auto/common/ratesheets/Canopius Phys Dam Rater.html",
    EmailFormalQuote: "toga/templates/commercial/auto/common/formalquote/formalquote.html",
    EmailPolicyBind: "toga/templates/commercial/auto/common/notifications/bindsuccessmailbody.html",
    EmailPolicyCancel: "toga/templates/commercial/auto/common/notifications/cancelsuccessmailbody.html",
    EmailPolicyReInstate: "toga/templates/commercial/auto/common/notifications/reinstatesuccessmailbody.html",
    TrueString: "true",
    FalseString: "false",
    PageChangeFormName: "pageChangeForm",
    New: "new",
    SubmitAndAddAnother: "submitAndAddAnother",
    SubmitOnly: "SubmitOnly",
    CloseModal: "closeModal",
    Completed: "completed",
    Errored: "errored",
    Deleted: "Deleted",
    DatatTableMode: "checkbox",
    AddNewDriver: "addnewDriver",
    Disabled: "disabled",
    Disable: "disable",
    Selected: "selected",
    Options: "options",
    FMCSA: "FMCSA",
    YesString: "Yes",
    NoString: "No",
    TemplateNameDeclineSubmission: "toga/templates/commercial/auto/common/notifications/submissiondecline.html",
    TemplateNameHoldSubmission: "toga/templates/commercial/auto/common/notifications/submissionhold.html",
    UnderwriterReferral: "UnderwriterReferral",
    String: 'string',
    TemplateNameFmcsaDocument: "diep2/templates/toga/templates/commercial/auto/common/fmcsa/businessinfo_bydotnumber.html",
    Individual: "Individual",
    LocationTypeArray: "array",
    Active: "Active"
}
const accountHubKeys = {
    AgencyName: "agentName",
    UWName: "underwriterName",
}

export {
    commonKeys,
    dateKeys,
    accountHubKeys
}