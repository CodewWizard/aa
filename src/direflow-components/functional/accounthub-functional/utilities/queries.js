import gql from 'graphql-tag';

export const GETACCOUNTHUB = gql`
query GetByFilter ($input: filters)
{
  getByFilter(input: $input) {
    id
    EffectiveDate
    ExpirationDate
    QuoteNumber
    PolicyNumber
    PolicyStatus
    Agency {
      Code
      Name
    }
    Agent {
      Code
      Name
    }
    UnderWriter {
      Code
      Name
    }
    InsuredAccount {
      BusinessInfo {
        BusinessName
        BusinessType
        FirstName
        MiddleName
        LastName
        Locations {
          Address {
            Number
            AddressType
            Description
            AddressLine1
            AddressLine2
            AptSuite
            City
            County
            CountyCode
            State
            PoBox
            Postalcode
            FormattedAddress
            UnFormattedAddress
            AdministrationArea1
            AdministrationArea2
            Locality
            Lat
            Long
            Street
            Name
            Premise
            Status
            Territory
            StreetName
            IsManual
            Business
            PlaceId
          }
          NickName
          LocationNumber
          IsValid
          Status
          Type
      }
      }
    }
    TotalPremium {
      AnnualPremium
    }
    Audit {
      CreatedBy
      CreatedOn
      LastUpdatedBy
      LastUpdatedOn
    }
    Transaction {
      Date
      Type
      Status
      Number
    }
    ExternalRefrences {
        ReferenceNumber
        ReferenceTarget
        Status
    }
    Attributes {
      AppSource
      isMaster
      Lob
    }
  }
}
`;