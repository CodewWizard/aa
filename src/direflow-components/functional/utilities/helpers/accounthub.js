import { callAPI } from "../../utilities/helpers/shared";

export const LoadNextPage = async (page, maxItemCount, maxSize, totalSize, base, query, ProcessAccountHubData) => {
    try {
        let input = {
            input: {
                containerName: "policy",
                page,
                maxItemCount,
                totalSize,
                maxSize,
                client: base.client
            },
        };

        const gqlpayload = JSON.stringify({
            operationName: "GetByPagination",
            query: typeof query === "string" ? query : query.loc.source.body,
            variables: input
        });

        const resp = await callAPI(base.gqlEP, "POST", gqlpayload, {
            "Authorization": `Bearer ${base.token}`,
            'Content-Type': 'application/json'
        });

        let data = [];
        if (resp) data = ProcessAccountHubData(resp.data["getByPagination"].policy);
        return { data: data || [], totalCnt: resp.data["getByPagination"].totalSize };
    } catch (error) {
        debugger
        console.log({ error });
    }
};