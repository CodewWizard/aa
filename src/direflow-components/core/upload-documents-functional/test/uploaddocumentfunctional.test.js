import React from "react";
import renderer, { act } from "react-test-renderer";
import { render, screen, cleanup, fireEvent, waitFor } from "@testing-library/react";
import UploadDocumentFunctional from "../uploaddocumentfunctional";
import * as utils from '../utilities/helpers'

jest.mock('../utilities/helpers.js', ()=>{
    return {
        getFilesListFromDMS: jest.fn().mockImplementation(()=>{
            return Promise.resolve([{TransactionType: "NEW", DocumentType: "Policy Document"}, {TransactionType: "Application", DocumentType: "Data Submission"}])
        }),
        uploadFilesToDMS: jest.fn().mockImplementation(()=>{
            return Promise.resolve(true);
        })
    }
})

afterEach(()=>{
    cleanup();    
})

const policy = {
    PolicyNumber: "mockPolicy",
    QuoteNumber: "mock",
    PolicyStatus: "mock",
    Transaction: {Type: "mock"},
    "Attributes": {
        "Client": "ROTHERT",
        "AppSource": "",
        "Carrier": "SD",
        "Lob": "HO3",
        "State": "OR",
        "Product": "SDRHHO3OR",
        "RaterVersion": "1.0.0.0",
        "Coverholder": "RH",
        "IsMasterPolicy": true,
        "MasterReferenceNumber": "",
        "IsTestPolicy": false,
        "NoOfClaims": "0",
        "IsCoverageSaved": true,
      },
}

const DocumentsColumns = [
    {
      dataField: 'TransactionType',
      text: 'TRANSACTION TYPE	',
      sort: false,
      headerStyle: () => {
        return { width: "17%" };
      }
    },
    {
      dataField: 'UserName',
      text: 'User',
      sort: false,
      headerStyle: () => {
        return { width: "17%", textAlign: "center" };
      },
      customFormatter: (data)=>data+"Test"
    },
    {
      dataField: 'DocumentType',
      text: 'DOCUMENT TYPE',
      sort: false,
      headerStyle: () => {
        return { width: "25%", textAlign: "center"  };
      },
    },
    {
      dataField: 'Uri',
      text: 'FileUrl',
      sort: false,
      headerStyle: () => {
        return { width: "20%" };
      },
      hidden: true
    },
  
];

const env = {
  apiBaseUrl: 'https://dev.cogitate.us/DIEP2.DMS.API/api/',
  getDocListUrl: 'GetPolicyDocList?',
  postDocListUrl: 'DMS/CreateDocument',
  dmsApiAuth: 'Basic dGVzdDp0ZXN0VXNlclBhc3N3b3Jk',
  dmsApiName: 'dms',
  dmsDefaultCoverHolder:'RH',
  dmsContainerName: 'dms',
}

const functionalAgrs = {
  policy: policy,
  tableCols: DocumentsColumns,
  groupColumnName: "TransactionType",
  dmsContainerName:"dms",
  useRefresh: true,
  EmailDocument:  ms => new Promise(res => setTimeout(res, 2000)),
  uploadFilesToDMS: async (...args)=>{ return await uploadFilesToDMS(...args ,env.apiBaseUrl+env.postDocListUrl, env.dmsApiAuth, env.dmsDefaultCoverHolder, env.dmsApiName)},
  getFilesListFromDMS: async (args)=>{
     return await getFilesListFromDMS(args, env.apiBaseUrl+env.getDocListUrl, env.dmsApiAuth);
  } 
};

it("should render given schema correctly", () => {
  const testRenderer = renderer.create(<UploadDocumentFunctional {...functionalAgrs} />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
  testRenderer.unmount();
});

// it("should call getFilesListFromDMS onload", () => {
//     const mockUtils = jest.spyOn(utils, "getFilesListFromDMS");
//     render(<UploadDocumentFunctional {...functionalAgrs} />);
//     expect(mockUtils).toHaveBeenCalled();
// });

it("should call getFilesListFromDMS onload and return array of files", async () => {
    const mockUtils = jest.spyOn(utils, "getFilesListFromDMS");
    mockUtils.mockResolvedValue([]);
    render(<UploadDocumentFunctional {...functionalAgrs} />);
    expect(typeof await mockUtils()).toBe(typeof [])
});

// it("should upload file if upload button is clicked", async () => {
//     render(<UploadDocumentFunctional {...functionalAgrs} />);
//     const uploadDocumentBtn = screen.getByText(/Upload Document/g);
//     fireEvent.click(uploadDocumentBtn);
//     await waitFor(() => {
//         const docType = screen.getAllByRole("textbox");
//         expect(docType[0].value).toBe("mockPolicy");
//         fireEvent.change(docType[1], { target: { value: 'Test value' } });
//         expect(docType[1].value).toBe('Test value');
//     });
// });


