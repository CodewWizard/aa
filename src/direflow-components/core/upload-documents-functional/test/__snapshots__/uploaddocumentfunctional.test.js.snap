// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render given schema correctly 1`] = `
Array [
  <div
    className="first-table position-relative"
  >
    <h3
      className="m-0"
    >
      Documents
       Â 
      <a
        className=""
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-arrows-rotate fa-lg "
          data-icon="arrows-rotate"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M105.1 202.6c7.7-21.8 20.2-42.3 37.8-59.8c62.5-62.5 163.8-62.5 226.3 0L386.3 160H336c-17.7 0-32 14.3-32 32s14.3 32 32 32H463.5c0 0 0 0 0 0h.4c17.7 0 32-14.3 32-32V64c0-17.7-14.3-32-32-32s-32 14.3-32 32v51.2L414.4 97.6c-87.5-87.5-229.3-87.5-316.8 0C73.2 122 55.6 150.7 44.8 181.4c-5.9 16.7 2.9 34.9 19.5 40.8s34.9-2.9 40.8-19.5zM39 289.3c-5 1.5-9.8 4.2-13.7 8.2c-4 4-6.7 8.8-8.1 14c-.3 1.2-.6 2.5-.8 3.8c-.3 1.7-.4 3.4-.4 5.1V448c0 17.7 14.3 32 32 32s32-14.3 32-32V396.9l17.6 17.5 0 0c87.5 87.4 229.3 87.4 316.7 0c24.4-24.4 42.1-53.1 52.9-83.7c5.9-16.7-2.9-34.9-19.5-40.8s-34.9 2.9-40.8 19.5c-7.7 21.8-20.2 42.3-37.8 59.8c-62.5 62.5-163.8 62.5-226.3 0l-.1-.1L125.6 352H176c17.7 0 32-14.3 32-32s-14.3-32-32-32H48.4c-1.6 0-3.2 .1-4.8 .3s-3.1 .5-4.6 1z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </h3>
    <div
      className="summaryEdit"
    >
      <a
        className="edit-2"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-upload fa-lg "
          data-icon="upload"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 109.3V352c0 17.7-14.3 32-32 32s-32-14.3-32-32V109.3l-73.4 73.4c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l128-128c12.5-12.5 32.8-12.5 45.3 0l128 128c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L288 109.3zM64 352H192c0 35.3 28.7 64 64 64s64-28.7 64-64H448c35.3 0 64 28.7 64 64v32c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V416c0-35.3 28.7-64 64-64zM432 456a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Upload Document
      </a>
    </div>
    <div
      className="w-100"
    >
      <div
        className="singleBox customDataTable"
      >
        <div
          className="react-bootstrap-table"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th
                  style={
                    Object {
                      "width": "17%",
                    }
                  }
                  tabIndex={0}
                >
                  TRANSACTION TYPE	
                </th>
                <th
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "17%",
                    }
                  }
                  tabIndex={0}
                >
                  User
                </th>
                <th
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "25%",
                    }
                  }
                  tabIndex={0}
                >
                  DOCUMENT TYPE
                </th>
                <th
                  className="expand-cell-header"
                  data-row-selection={true}
                  onClick={[Function]}
                >
                  (+)
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  className="react-bs-table-no-data"
                  colSpan={4}
                  data-toggle="collapse"
                >
                  No Results Found !!!
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  <a
    alt="filedownload"
    id="filedownload"
  />,
]
`;
